next/src/modules/dashboard/components/ecg/MainChart.tsx

```diff
@@ -81,7 +82,13 @@ const MainChart: React.ForwardRefRenderFunction<HTMLDivElement, Props> = (
   const { theme, isDarkMode } = useTheme();
   const LEGEND_DATA = isDarkMode ? LEGEND_DATA_DARK : LEGEND_DATA_LIGHT;
 
-  const chartSettings = getChartSettings(theme);
+  const chartSettings = useMemo(
+    () =>
+      DATA_PROBLEM === 'ecg_classification'
+        ? getChartSettings(theme, mockEcgRanges, chartData?.data?.labels)
+        : getChartSettings(theme),
+    [theme, chartData],
```

Comment:
```suggestion
      if (DATA_PROBLEM === 'ecg_segmentation') {
        return getSegmentationChartSettings(theme, mockEcgRanges, chartData?.data?.labels)
      }
      if (DATA_PROBLEM === 'ecg_classification') {
        return getClassificationChartSettings(theme)
      }
    ...
```

I think that the chart settings shouldn't be shared between data problems.

Reply:
Should I add new data problem? `ecg_segmentation `. I was thinking that its improvements for ecg_classification.

Reply:
Yes that's the idea, to support multiple different tasks rather than modify a single one. Sorry, I thought it was obvious.